### PR TITLE
Im2vec in Tensorflow

### PR DESCRIPTION
**Pull request by agude**
This implements a (preliminary) Single Layer Perceptron (or not-so-deep neural net if you prefer) to translate image features into a text space.

### CODE REVIEW

**Comment by agude** [comment]
This needs to be updated to use the dataset iterators instead of loading data off the file system.

File: attalos/imgtxt_algorithms/sampledSM/train_tensorflow.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+# Constants
+batch_size = 128
+hidden_layer_size = 100
+learning_rate = 0.001
+negative_sample_count = 10
+training_epochs = 10
+
+# Read in the data
+train_image_feature_file = '/local_data/features/image/iaprtc12_train_inception.hdf5'
```
Comment (agude) [inline attalos/imgtxt_algorithms/sampledSM/train_tensorflow.py]: We don't want to hardcode paths; is the right answer to build a `main()` that reads from the command line?

File: attalos/imgtxt_algorithms/sampledSM/train_tensorflow.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        total = 0
+        for i in range(len(model_predicted_indices)):
+            total += len(np.intersect1d(model_predicted_indices[i], true_indices[i])) / 5.
+
+        total /= float(len(model_predicted_indices))
+        print("\tAccuracy: {:.2%}\n".format(total))
+
+        if not epoch % 100:
+            # Save the variables to disk.
+            save_path = saver.save(sess, "./my_model", global_step=epoch)
```
Comment (agude) [inline attalos/imgtxt_algorithms/sampledSM/train_tensorflow.py]: Likewise the save location and name should probably be something specified in arguments.

**Comment by ymt123** [comment]
Looks good!

**Pull request merged**